clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -314,50 +315,51 @@ class PointerVariableConstraint : public ConstraintVariable {
+  void brainTransplant(ConstraintVariable *From, ProgramInfo &I) override;
+  void mergeDeclaration(ConstraintVariable *From, ProgramInfo &I) override;
 
   static bool classof(const ConstraintVariable *S) {
     return S->getKind() == PointerVariable;
   }
 
   std::string mkString(EnvironmentMap &E, bool EmitName =true,
-                       bool ForItype =false, bool EmitPointee = false) const;
+                       bool ForItype = false,
+                       bool EmitPointee = false) const override;
 
   FunctionVariableConstraint *getFV() const { return FV; }
 
-  void print(llvm::raw_ostream &O) const ;
-  void dump() const { print(llvm::errs()); }
-  void dump_json(llvm::raw_ostream &O) const;
-  void constrainToWild(Constraints &CS) const;
-  void constrainToWild(Constraints &CS, const std::string &Rsn) const;
+  void print(llvm::raw_ostream &O) const override ;
+  void dump() const override { print(llvm::errs()); }
+  void dump_json(llvm::raw_ostream &O) const override;
+  void constrainToWild(Constraints &CS) const override;
+  void constrainToWild(Constraints &CS, const std::string &Rsn) const override;
   void constrainToWild(Constraints &CS, const std::string &Rsn,
-                       PersistentSourceLoc *PL) const;
+                       PersistentSourceLoc *PL) const override;
   void constrainOuterTo(Constraints &CS, ConstAtom *C, bool doLB = false);
-  bool anyChanges(EnvironmentMap &E) const;
+  bool anyChanges(EnvironmentMap &E) const override;
   bool anyArgumentIsWild(EnvironmentMap &E);
-  bool hasWild(EnvironmentMap &E, int AIdx = -1) const;
-  bool hasArr(EnvironmentMap &E, int AIdx = -1) const;
-  bool hasNtArr(EnvironmentMap &E, int AIdx = -1) const;
+  bool hasWild(EnvironmentMap &E, int AIdx = -1) const override;
+  bool hasArr(EnvironmentMap &E, int AIdx = -1) const override;
+  bool hasNtArr(EnvironmentMap &E, int AIdx = -1) const override;
 
-  void equateArgumentConstraints(ProgramInfo &I);
+  void equateArgumentConstraints(ProgramInfo &I) override;
 
   bool isPartOfFunctionPrototype() const  { return partOFFuncPrototype; }
   // Add the provided constraint variable as an argument constraint.
   bool addArgumentConstraint(ConstraintVariable *DstCons, ProgramInfo &Info);
   // Get the set of constraint variables corresponding to the arguments.
   const std::set<ConstraintVariable *> &getArgumentConstraints() const;
 
-  bool isEmpty(void) const { return vars.size() == 0; }
+  bool isEmpty(void) const override { return vars.size() == 0; }
 
-  ConstraintVariable *getCopy(Constraints &CS);
+  ConstraintVariable *getCopy(Constraints &CS) override;
 
   // Retrieve the atom at the specified index. This function includes special
   // handling for generic constraint variables to create deeper pointers as
   // they are needed.
   Atom *getAtom(unsigned int AtomIdx, Constraints &CS);
 
-  virtual ~PointerVariableConstraint() {};
+  ~PointerVariableConstraint() override {};
```

Comment:
Why `virtual` dropped? Is it weird to have it for a constructor?

Reply:
Just a recommendation I saw somewhere. I think override implies virtual, but is stronger because there must be corresponding virtual method in the super class.